src/coverage.ts

```diff
@@ -164,14 +165,24 @@ async function publishCoverage(
           console.log(data);
           continue;
         }
-        const response = await fetch(opts.url, {
-          method: 'POST',
-          body: JSON.stringify(data),
-          headers: { Authorization: `Bearer ${opts.token}`, 'Content-Type': 'application/json' }
-        });
-
-        if (!response.ok) {
-          throw new Error(`Failed to publish coverage: ${response.status} ${response.statusText}`);
+        const body = JSON.stringify(data);
+        let attempts = 0;
+        let done = false;
+        while (!done) {
+          const response = await fetch(opts.url, {
+            method: 'POST',
+            body,
+            headers: { Authorization: `Bearer ${opts.token}`, 'Content-Type': 'application/json' }
+          });
+          if (response.ok) {
```

Comment:
We retry not just connection errors and others, but also 4xx and 3xx which isn't quite correct and could cause further issues.

Additionally, shouldn't there be a back-off to make the retry more likely to succeed?

Reply:
I recall seeing at least some transient 4xxs from codestats. For 3xxs not sure what to do but follow them in case we do not yet. Can check.

For backoff, maaybe. not sure how long I would be willing to wait. 1 - 2 - 4 seconds?

Reply:
I added simple backoff

Reply:
The behaviour on 4xx and redirects would be surprising for anyone using this action that doesn't have the same backend